scripts/ec2/deploy_tags.coffee

```diff
@@ -20,65 +22,93 @@ RESERVE_TAGS = {
   description: 'ReservationDescription',
 }
 
-reserveForDeploy = (msg, instance, reservation) ->
-  tags.addReservation(msg, instance, reservation)
-  msg.send "Reservation added to #{instance}."
-
-getReservationTags = (args) ->
-  reservationUser = ///--#{RESERVE_TAGS.user}=(.*?)(\s+|$)///.exec(args)[1]
-  reservationTime = Date.now().toString()
-  reservationBranch = ///--#{RESERVE_TAGS.branch}=(.*?)(\s+|$)///.exec(args)[1]
-  reservationDescription = ///--#{RESERVE_TAGS.description}="(.*?)"///.exec(args)[1]
-  [
-    {Key: RESERVE_TAGS.user, Value: reservationUser},
-    {Key: RESERVE_TAGS.time, Value: reservationTime},
-    {Key: RESERVE_TAGS.branch, Value: reservationBranch},
-    {Key: RESERVE_TAGS.description, Value: reservationDescription},
-  ]
+RESERVE_CONFIG_ENV_VAR = "HUBOT_AWS_RESERVE_CONFIG"
+
+try
+  INSTANCE_ID_MAP = cson.parseCSONFile process.env[RESERVE_CONFIG_ENV_VAR]
+catch e
+  # TODO switch to `robot.logger()` here _without_ constantly reparsing config file
+  if e.name == "TypeError"
+    console.log "#{RESERVE_CONFIG_ENV_VAR} is not defined"
+  else
+    console.log "#{RESERVE_CONFIG_ENV_VAR} could not be loaded " +
+                "(#{process.env.HUBOT_AWS_RESERVE_CONFIG})"
+  INSTANCE_ID_MAP = {}
+
+
+getReservationTags = (msg) ->
+  args = if msg.match[1] then msg.match[1].trim().split(/\s+/).slice(1) else []
+  if args.length < 2
+    []
+  else
+    [
+      {Key: RESERVE_TAGS.user, Value: msg.message.user.name},
+      {Key: RESERVE_TAGS.time, Value: Date.now().toString()},
+      {Key: RESERVE_TAGS.branch, Value: args[0]},
+      {Key: RESERVE_TAGS.description, Value: args.slice(1).join(" ")},
+    ]
 
 formatReservationsList = (instances) ->
   if not instances.length
     return '[None]'
 
   getTag = (_tags, tag) -> _.first(t.Value for t in _tags when t.Key is tag) || ''
 
+  instanceMap = _.invert(INSTANCE_ID_MAP)
+
   rows = (
     {
-      id: i.InstanceId,
-      name: getTag(i.Tags, 'Name'),
-      ip: i.PrivateIpAddress,
-      state: i.State.Name,
+      id: instanceMap[i.InstanceId] || "UNKNOWN",
       resUser: getTag(i.Tags, RESERVE_TAGS.user),
       resTime: getTag(i.Tags, RESERVE_TAGS.time),
       resBranch: getTag(i.Tags, RESERVE_TAGS.branch),
-      resDescrip: getTag(i.Tags, RESERVE_TAGS.description)
+      resDescrip: getTag(i.Tags, RESERVE_TAGS.description),
```

Comment:
apologies for the mix of parens and no-parens when calling functions – coffeescript 🙄